tests/001-handshake.cpp

```diff
@@ -93,29 +93,104 @@ namespace oxen::quic::test
             CHECK_FALSE(public_ipv6.is_loopback());
         };
 
-        SECTION("Endpoint object creation - Default addressing")
+        SECTION("IP Address Ranges", "[range][operators][ipaddr]")
         {
-            Network test_net{};
-            Address default_addr{};
+            CHECK((ipv4(10, 0, 0, 0) / 8).contains(ipv4(10, 0, 0, 0)));
+            CHECK((ipv4(10, 0, 0, 0) / 8).contains(ipv4(10, 255, 255, 255)));
+            CHECK((ipv4(10, 123, 45, 67) / 8).contains(ipv4(10, 123, 123, 123)));
+            CHECK((ipv4(10, 255, 255, 255) / 8).contains(ipv4(10, 0, 0, 0)));
+            CHECK((ipv4(10, 255, 255, 255) / 8).contains(ipv4(10, 123, 123, 123)));
+            CHECK_FALSE((ipv4(10, 0, 0, 0) / 8).contains(ipv4(11, 0, 0, 0)));
+            CHECK_FALSE((ipv4(10, 0, 0, 0) / 8).contains(ipv4(9, 255, 255, 255)));
+
+            CHECK((ipv6(0x2001, 0xdb8) / 32).contains(ipv6(0x2001, 0xdb8)));
+            CHECK((ipv6(0x2001, 0xdb8) / 32).contains(ipv6(0x2001, 0xdb8, 0xffff, 0xffff)));
+            CHECK((ipv6(0x2001, 0xdb8, 0xffff) / 32).contains(ipv6(0x2001, 0xdb8)));
+            CHECK((ipv6(0x2001, 0xdb8, 0xffff) / 32).contains(ipv6(0x2001, 0xdb8)));
+        }
 
-            auto ep = test_net.endpoint(default_addr);
-            // Note: kernel chooses a random port after being passed default addr
-            CHECK_FALSE(ep->local().to_string() == default_addr.to_string());
-        };
+#ifndef _WIN32
+        SECTION("IPv4 Addresses", "[ipv4][constructors][ipaddr]")
+        {
+            uint32_t v4_n;                    // network order ipv4 addr
+            auto v4_h = "192.168.1.1"s;       // host order ipv4 string
+            auto v4_full = "192.168.1.1:0"s;  // full ipv4 addr/port string
```

Comment:
This could be a non-zero port to have symmetry with the ipv6 section after this.

Reply:
valid, changed :+1: